app/src/main/java/ch/onepass/onepass/utils/TimeProviderHolder.kt

```
@@ -0,0 +1,10 @@
+package ch.onepass.onepass.utils
+
+object TimeProviderHolder {
+  lateinit var instance: TimeProvider
```

The current `TimeProviderHolder` implementation uses a `lateinit` variable to store the `TimeProvider` instance. This introduces a potential race condition: if `instance` is accessed before `initialize()` is called, the app will crash with an `UninitializedPropertyAccessException`.  

To make this safer, it’s recommended to guard access to the variable and fail fast with a clear error message:

```kotlin
val instance: TimeProvider
  get() = if (::instance.isInitialized) instance
          else error("TimeProviderHolder not initialized")
````

This approach prevents unpredictable crashes and makes it immediately obvious during development if the instance is being accessed too early.

---

You're right - better to fail fast with a clear message than crash mysteriously. Added a safe getter.